broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js

```diff
@@ -414,12 +414,75 @@ class ServiceBrokerApiController extends FabrikBaseController {
       .catch(NotFound, notFound);
   }
 
+  getLastBindingOperation(req, res) {
+    const encodedOp = _.get(req, 'query.operation', undefined);
+    const operation = encodedOp === undefined ? {} : decodeBase64(encodedOp);
+    const guid = req.params.binding_id;
+    const namespaceId = eventmesh.apiServerClient.getNamespaceId(req.params.instance_id);
+    let action, instanceType;
+
+    function done(result) {
+      const body = _.pick(result, 'state', 'description');
+      if (body.state === CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS ||
```

Comment:
Is delete also needed here?

Reply:
I am observing state as `succeeded` after calling async delete binding request in SFServiceBindings.
Below is an Sfservicebindings CRD which is in delete state,
```yaml
Name:         726d9304-c24e-4edf-a6ec-5d4137aea793
Namespace:    sf-968bc875-df93-4edd-8e9c-9a4e970141e3
Labels:       OWNER_INTEROPERATOR_NAMESPACE=interoperator
              instance_guid=968bc875-df93-4edd-8e9c-9a4e970141e3
              interoperator.servicefabrik.io/lastoperation=delete
              state=delete
Annotations:  <none>
API Version:  osb.servicefabrik.io/v1alpha1
Kind:         SFServiceBinding
Metadata:
  Creation Timestamp:             2021-04-22T11:21:23Z
  Deletion Grace Period Seconds:  0
  Deletion Timestamp:             2021-04-22T11:21:40Z
  Finalizers:
    broker.servicefabrik.io
    
    
 Status:
  Applied Spec:
    Instance Id:  
    Plan Id:      
    Service Id:   
  Response:
    Secret Ref:  sf-726d9304-c24e-4edf-a6ec-5d4137aea793
  State:         succeeded
  ```
  We are fetching `State` from `Status`. So I felt we will not have state as delete. The next lastoperation call will remove the finalizer and delete this CRD.

Reply:
I took a closer look. We are indeed setting the `State` to `delete`. Here is a log from broker,
```
2021-04-22T18:49:44.703Z - info: RequestIdentity: Absent , Updating - Resource 58c39b85-c62b-493a-87e7-637e05d89074 with body - {"metadata":{"labels":{"state":"delete"}},"status":{"state":"delete","description":""}}
```
We are requesting update to [ApiClient](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/broker/data-access-layer/eventmesh/src/ApiServerClient.js#L1072) but this change is not reflected in CRD.
This mismatch is also seen in async deprovisioning of service instance.

Reply:
@vivekzhere, You are right. I didn't realized that the interoperator is acting upon the CRD and changing the state to `succeeded`.
```bash
kubectl -n sf-10b19e82-7ea6-4513-a6f8-2f341b8354f9 get SFserviceinstances 10b19e82-7ea6-4513-a6f8-2f341b8354f9 -w
NAME                                   STATE         AGE   CLUSTERID
10b19e82-7ea6-4513-a6f8-2f341b8354f9   in progress   57s   1
10b19e82-7ea6-4513-a6f8-2f341b8354f9   succeeded     58s   1
10b19e82-7ea6-4513-a6f8-2f341b8354f9   succeeded     106s   1
10b19e82-7ea6-4513-a6f8-2f341b8354f9   delete        106s   1
```
When the state is in `delete`, we are getting the response of lastoperation as,
```json
{
    "state": "delete",
    "description": ""
}
```
But we are expecting a response with state as `in progress`. So I will add the condition suggested by you.
```suggestion
      if (body.state === CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS ||  body.state === CONST.APISERVER.RESOURCE_STATE.DELETE
```
Thank you 👍